File: frontend/src/app/models/event.ts
import { Person } from "./person";

export class Event {
    id: number;
    participants: Person[] = [];
    name: string;
    maxParticipants: number;
    startTime: Date;
    endTime: Date;
  
    constructor(id: number, participants: Person[], name: string, maxParticipants: number, startTime: Date, endTime: Date) {
      this.id = id;
      this.participants = participants;
      this.name = name;
      this.maxParticipants = maxParticipants;
      this.startTime = startTime;
      this.endTime = endTime;
    }
  }

File: backend/src/main/java/com/ng/events/model/Event.java
package com.ng.events.model;

import java.time.LocalDateTime;
import java.util.List;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinTable;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToMany;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Event {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToMany
    @JoinTable(
        name = "event_person",
        joinColumns = @JoinColumn(name = "event_id"),
        inverseJoinColumns = @JoinColumn(name = "person_id")
    )
    private List<Person> participants;

    private String name;

    private Long maxParticipants;

    private LocalDateTime startTime;

    private LocalDateTime endTime;
    
}
